src/array/cpu/labor_pick.h

```diff
@@ -45,6 +45,13 @@ namespace impl {
 
 using dgl::random::continuous_seed;
 
+template <typename K, typename V>
+using map_t = tsl::robin_map<K, V>;
+template <typename iterator>
+auto& mutable_value_ref(iterator it) {
```

Comment:
why wrap the value access with a function? Is it necessary? It's different from std::unordered_map?

Reply:
Yes, it->second is const with this map implementation by default. Use .value() to access it. I wrapped it so that when we change map again for any reason, we have to only change this function.